src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -2986,8 +2986,8 @@ public async Task QueryExpression3()
     int Method()
     {
         var q = from c in Get(1 +
-                            2 +
-                            3)
+                                2 +
+                                3)
```

Comment:
i'm curious what happened here.

Reply:
These lines previously anchored to `from`, but now anchor to the first token on the line containing `from` (`var`). Since `var` didn't move, these lines were not altered.

Reply:
Note that anchoring is different from relative indentation. The latter applies to the `select` keyword on the next line.

Reply:
hrmmmmmm. 

i am worried about this.  are we sure we don't want this to be anchored to the 'from'.  While i genuinely think the anchoring change makes sense for many cases, queries are *so* special that it seems like we would want them them to follow that.  (lambdas may be similar in that regard).

Reply:
Also note that the new behavior is only problematic when all of the following hold:

1. The original anchor token is not the first token on the line
2. The first token on the line with the anchor token is in the correct location
3. The anchor token is not in the correct location (i.e. spaces need to be added or removed to correct the position of the anchor token)
4. The now position of the anchor token continues to be on the same line
5. The anchored token on a subsequent line is correctly aligned with the mispositioned anchor token, and needs to retain its position after the anchor adjustment

Of all the tests that needed updating, this is the only one which could be argued to meet all these conditions. However, it's also easy to argue that this case does not meet condition (5), and therefore the new behavior is equally acceptable.

Reply:
➡️ Testing greatly increased by adding an idempotency check for format document tests